Tests/Sources/VLCLibraryTest.swift

```diff
@@ -0,0 +1,119 @@
+ * $Id$
+ *
+ * Authors: Mike JS. Choi <mkchoi212 # icloud.com>
+ *
+ * This program is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation; either version 2.1 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston MA 02110-1301, USA.
+ *****************************************************************************/
+
+import XCTest
+
+class VLCLibraryTest: XCTestCase {
+    
+    let paramKey = "VLCParams"
+    
+    override func setUp() {
+        super.setUp()
+        UserDefaults.standard.removeObject(forKey: paramKey)
+    }
+
+    func testSharedLibrary() {
+        XCTAssertNotNil(VLCLibrary.shared())
+        XCTAssertNotNil(VLCLibrary.shared().instance)
+    }
+    
+    func testInitWithOptions() throws {
+        let tests = [
+            ["--no-video-title-show", "--verbose=4"],
+            []
+        ]
+        
+        for option in tests {
+            let library = try XCTAssertNotNilAndUnwrap(VLCLibrary(options: option))
+            XCTAssertNotNil(library.instance)
+            assertDefaultParameters()
+        }
+    }
+    
+    func testInit() {
+        let library = VLCLibrary()
+        XCTAssertNotNil(library.instance)
+        assertDefaultParameters()
+    }
+    
+    func testDebugLoggingLevel() throws {
+        let library = try XCTAssertNotNilAndUnwrap(VLCLibrary.shared())
+        XCTAssertEqual(library.debugLoggingLevel, 0)
+        
+        let tests: [(input: Int32, expected: Int32)] = [
+            (0, 0),
+            (3, 3),
+            (4, 4),
+            (100, 0),
+            (-10, 0)
+        ]
+        
+        for (input, expected) in tests {
+            library.debugLoggingLevel = input
+            XCTAssertEqual(library.debugLoggingLevel, expected)
+        }
+    }
+    
+    func testDebugLogging() throws {
+        let library = try XCTAssertNotNilAndUnwrap(VLCLibrary.shared())
+        XCTAssertFalse(library.debugLogging)
+        
+        library.debugLogging = true
+        XCTAssertTrue(library.debugLogging)
+        
+        library.debugLogging = false
+        XCTAssertFalse(library.debugLogging)
+    }
+    
+    func testLibraryDescription() throws {
+        let library = try XCTAssertNotNilAndUnwrap(VLCLibrary.shared())
+        
+        XCTAssert(library.version.count > 1)
+        XCTAssert(library.compiler.count > 1)
+        XCTAssert(library.changeset.count > 1)
```

Comment:
What you're testing for here is a bit unclear for me. Those are strings but the way the test is written it looks like you're testing an array.  What do you think about testing for a not empty string and providing a message like "compiler should hold the string of the libvlccompiler, e.g. "gcc version 4.2.3 (Ubuntu 4.2.3-2ubuntu6)"? That way if the test fails we know roughly what we expected

Reply:
I thought about doing 
```swift
XCTAssertEqual(library.compiler, "gcc version 4.2.3 (Ubuntu 4.2.3-2ubuntu6))"
```
but every time the library gets an update, you would have to update the tests as well or else, they would fail. Is this desirable?

Reply:
I was thinking more along the lines of XCTAssertFalse (library.compiler. isEmpty, "compiler is empty but it should hold a string of the form e.g. "gcc version 4.2.3 (Ubuntu 4.2.3-2ubuntu6)")